### PR TITLE
LPD-39598 Refactor of poshi test

### PR DESCRIPTION
**Pull request by monaco-fabio**
https://liferay.atlassian.net/browse/LPD-39598

### CODE REVIEW

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:playwright:nocompile

File: modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts [hunk 1]
```diff
@@ -756,3 +756,237 @@ test(`LPD-29993 Users can view and download a product's attachments`, async ({
...
+	);
+	const accountRoleBuyer = rolesResponse?.items?.filter((role) => {
+		return role.name === 'Buyer';
+	});
+
+	await apiHelpers.headlessAdminUser.assignAccountRoles(
+		account.externalReferenceCode,
+		accountRoleBuyer[0].id,
+		user.emailAddress
+	);
```
Comment (andrea-ale-sbarra) [inline modules/test/playwright/tests/commerce/commerce-product-content-web/productDetails.spec.ts:808]: move these lines at the beginning of the test, the idea is to use variables close to where you declared it, if you create a catalog and the catalog is used after the account creation you are breaking the logical chain..

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:playwright:nocompile

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**